honeybee_vtk/writer.py

```diff
@@ -57,35 +93,69 @@ def writer(file_path, *, file_name=None, target_folder=None, include_grids=False
+def writer(file_path, *, file_name=None, target_folder=None, writer='html',
+           open_html=False, include_grids=False, include_sensors=False,
+           include_normals=False):
+    """Read a valid HBJSON and either write an HTML or a .zip of VTK files.
 
-        elif writer.lower() == 'xml':
-            vtk_writer = vtk.vtkXMLPolyDataWriter()
-            vtk_extension = '.vtp'
-            return write_files(
-                hbjson, file_path, file_name, target_folder, include_grids,
-                include_vectors, include_points, vtk_writer, vtk_extension)
+    Args:
+        file_path: A text string for a valid path to the HBJSON file.
+        file_name: A text string for the name of the .zip file to be written. If no
+            text string is provided, the name of the HBJSON file will be used as a
+            file name for the .zip file.
+        target_folder: A text string to a folder to write the output file. The file
+            will be written to the current folder if not provided.
+        writer: A text string to indicate the VTK writer. Acceptable values are
+            'vtk', 'xml', and 'html'. Defaults to 'html'.
+        open_html: A boolean. If set to True, it will open the generated HTML
+            in a web browser when 'html' is provided as value in the writer argument.
+            Defaults to False.
+        include_grids: A boolean. Grids will be exported if the value is True. Here, a
+            grid is exported as base-geometry, grid mesh, or sensorpoints in that order
+            of preference. So if a Sensorgrid object in HBJSON has base-geometry, then
+            that object will be exported as base geometry, eventhough it also might have
+            mesh and sensors.
+        include_sensors: A text string to indicate whether to show sensor directions as
+            vectors or points colored based on directions. Acceptable values are;
+            'vectors' and 'points.' Defaults to False.
+        include_normals: A text string to indicate whether to show face normals as
+            vectors or points colored based on directions. Acceptable values are;
+            'vectors' and 'points.' Defaults to False. Currently, this function
+            only exports normals for the apertures. Other face types will be supported
+            in future.
 
-        elif writer.lower() == 'vtk':
-            vtk_writer = vtk.vtkPolyDataWriter()
-            vtk_extension = '.vtk'
-            return write_files(
-                hbjson, file_path, file_name, target_folder, include_grids,
-                include_vectors, include_points, vtk_writer, vtk_extension)
+    Returns:
+        A text string containing the path to the output file.
+    """
 
-        else:
-            raise ValueError(writer_error)
+    hbjson, writer_val = _check_inputs(
+        file_path, file_name, target_folder, writer, open_html, include_grids,
+        include_sensors, include_normals)
```

Comment:
I think these inputs should be validated in the functions that you are calling below. This extra checks here look unnecessary to me. I'll just remove it and make sure the functions below are self-sufficient to report errors.